### ISSUE TITLE
Testing ContentDB on non-Linux platforms

### ISSUE THREAD

**Issue by rubenwardy** [minetest/minetest#8165]
I can't easily test on platforms other than Linux

1. Install random shit from CDB (check that buttons and the content tab update without restart)
2. Rename modpacks (check that buttons and the content tab update without restart)
3. Enable mods and modpacks, check that the correct things appear in world.mt and game
4. delete things from content tab (check that buttons and the content tab update without restart)
5. uninstall things from contentdb dialog (check that buttons and the content tab update without restart)
6. test updating
    1. Go into mod.conf / game.conf / modpack.conf / etc of a thing, and change "release" to a lower number
    2. Go to the ContentDB menu, find the thing, chack that the button changes from "uninstall" to "update".
    3. Click update
    4. check that the button changes back from "update" to uninstall without a restart

**Comment by adrido** [comment]
Works fine on Windows x64 Installer (Non run in place) also installing to the correct location, does even  create the folders if they dont exist, except this 3 minor things:
1. Installing a game does not make it appear on the game bar, only after a restart.
2. Minetest Game is not recognized by the Content Db as installed (appears on the content tab), can be installed twice, and removed once. This could be also a Issue of the installler, because it installs the Minetest Game to C:/Program Files/Minetest/games too. To uninstall the installer have to be used.
3. Clicking "Browse online content" for the first time freezes Minetest for a long time. (I have a very worse internet connection) Suggestion: Display a progress bar.

Well done 👍

**Comment by NathanSalapat** [comment]
Installing games has the same problem on Linux, or does for me at least.

**Comment by rubenwardy** [comment]
Are you sure you have the most recent version? The first of those should be fixed

**Comment by random-geek** [comment]
Apparently I did not. I've updated the post.

**Comment by Fixer-007** [comment]
Type Kek in search, press >>, error:
```
ERROR[Main]: Main menu error: Runtime error from mod '' in callback handleMainMenuButtons(): ...\minetest\bin\..\builtin\mainmenu\dlg_contentstore.lua:155: attempt to index local 'package' (a nil value)
ERROR[Main]: stack traceback:
ERROR[Main]: 	...\minetest\bin\..\builtin\mainmenu\dlg_contentstore.lua:155: in function 'get_screenshot'
ERROR[Main]: 	...\minetest\bin\..\builtin\mainmenu\dlg_contentstore.lua:463: in function 'get_formspec'
ERROR[Main]: 	...\minetest\bin\..\builtin\mainmenu\dlg_contentstore.lua:540: in function 'handle_buttons'
ERROR[Main]: 	blah\minetest\bin\..\builtin\fstk\ui.lua:148: in function 'handle_buttons'
ERROR[Main]: 	blah\minetest\bin\..\builtin\fstk\ui.lua:192: in function <blah\minetest\bin\..\builtin\fstk\ui.lua:178>
```

**Comment by Fixer-007** [comment]
Uninstalling that NodeCore game via general uninstall and CDB uninstall leaves NodeCore game Icon until game restart.

**Comment by paramat** [comment]
@rubenwardy is there anything else here, other than #8244 , that is a blocker?

**Comment by paramat** [comment]
#8244 and #8192 merged.

**Comment by paramat** [comment]
Remaining issues are:

> Minetest Game is not recognized by the Content Db as installed (appears on the content tab), can be installed twice, and removed once. This could be also a Issue of the installler, because it installs the Minetest Game to C:/Program Files/Minetest/games too. To uninstall the installer have to be used.

> Related: Enabled modpacks shown in green in the world configuration menu, but the it goes back to gray after exiting the menu and re-entering.

> Low priority: After uninstalling a mod, the cursor in the world configuration menu may move down several entries, causing a modpack to be expanded. Installing mods also causes this weirdness.

All seem non-blockers.

**Comment by pauloue** [comment]
I think the second one is fixed (https://github.com/minetest/minetest/commit/a809f7361e8a7232b3c9ec241be5adcae96b3b2d).

**Comment by Fixer-007** [comment]
My error was fixed as well, ty.

**Comment by paramat** [comment]
Yes i thought grey modpacks was fixed but on testing it doesn't seem to be.
